### PR TITLE
Make .editorconfig configurable.

### PR DESCRIPTION
**Pull request by knpwrs**
Continuation of #8. Adds configuration options to choose `2 spaces`, `4 spaces`, or `tabs` for indentation, and allows people to specify if they want to trim whitespace from markdown files.

### CODE REVIEW

File: sao.js [hunk 1]
```diff
@@ -107,6 +107,17 @@ module.exports = {
           ? true
           : 'Please include a valid GitHub.com URL without a trailing slash';
       }
+    },
+    trimWhitespaceMarkdown: {
+      message: 'Trim trailing whitespace in markdown files?',
```
Comment (niftylettuce) [inline sao.js]: You don't need to append `?` in the string, it is prefixed.

Btw if you want to test your fork, you can do so locally by running `./cli.js some-folder-name`.

File: sao.js [hunk 2]
```diff
@@ -107,6 +107,17 @@ module.exports = {
...
+      type: 'confirm',
+      default: true
+    },
+    editorconfigIndentation: {
+      message: 'Indentation',
+      type: 'list',
+      choices: ['2 spaces', '4 spaces', 'tabs'],
+      default: '2 spaces'
```
Comment (niftylettuce) [inline sao.js]: Curious if people have a global `~/.editorconfig` or something?  Not sure if htat's a standard, if so we could make default whatever value they use globally?

File: template/.editorconfig [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
+indent_size = 2<% } else if (editorconfigIndentation === '4 spaces') { %>indent_style = space
+indent_size = 4<% } else if (editorconfigIndentation === 'tabs') { %>indent_style = tab<% } %>
 end_of_line = lf
 charset = utf-8
 trim_trailing_whitespace = true
-insert_final_newline = true
+insert_final_newline = true<% if (!trimWhitespaceMarkdown) { %>
+
+[*.md]
+trim_trailing_whitespace = false<% } %>
```
Comment (niftylettuce) [inline template/.editorconfig]: Good stuff 👍

File: sao.js (hunk 1 above)
Comment (knpwrs) [reply sao.js]: I was testing with `npm link`. I could have sworn I saw some lines that had question marks. I'll remove it.

File: sao.js (hunk 2 above)
Comment (knpwrs) [reply sao.js]: I'm not personally aware of such a practice. I think that would assume there be tooling that looks for global `.editorconfig` files which could break the paradigm of configuring editors per project (maybe).

Comment (niftylettuce) [reply sao.js]: Yeah, lets not worry about it

**Comment by niftylettuce** [comment]
@knpwrs My only other comment is that you probably want to take into consideration tabs/spaces in the `xo` configuration object of `package.json`.

**Comment by knpwrs** [comment]
I updated the template for `package.json` but there's a problem. When the indentation is set to four spaces the `postinstall` script fails because the files have two spaces. Oddly, when the indentation is set to tabs I just leave that line out of the `xo` configuration and the `postinstall` script doesn't fail.

```json
"spaces": 4,
```

Results in:

image

And leaving out the spaces line results in a passing build, even though the files still have two spaces.

Also with the changes to `package.json` you have to force commit (`--no-verify`) now:

image

I've force committed and pushed the commit as-is for now.

**Comment by knpwrs** [comment]
So I can think of two solutions for the git hook:

1. Don't lint `template/package.json`.
1. Rename `template/package.json` to `template/package.json.ejs` and have `sao` move it.

The tab indentation thing is weird though...

**Comment by knpwrs** [comment]
Just pushed the rename solution, not sure about the four spaces vs tabs issue.

**Comment by OmgImAlexis** [comment]
@knpwrs can you not just get `xo` to fix the issue with tabs/spaces by setting the config and then running `xo --fix`?

**Comment by knpwrs** [comment]
There were a few solutions discussed on slack, I just need to find the time this weekend to put it together.

**Comment by OmgImAlexis** [comment]
Looks like it may be a bug with `xo`.

Removing this rule allows `xo --fix` to work properly.

```
      "prettier/prettier": [
        "error",
        {
          "singleQuote": true,
          "bracketSpacing": true,
          "trailingComma": "none"
        }
      ],
```

**Comment by OmgImAlexis** [comment]
From what I've found it looks like you can remove that section, run `xo --fix` and then add it back and everything gets converted to the amount of spaces you want.

**Comment by niftylettuce** [comment]
What should we do here?  I see they closed #252 and Prettier is strict... I guess we should drop Prettier due to all the issues?

**Comment by OmgImAlexis** [comment]
I'm good with dropping it, most of the standards it adds `xo` can help with anyways.

**Comment by niftylettuce** [comment]
@knpwrs feel free to open a new PR - there have been significant changes in the new release

**Pull request closed without merging**